File: project_euler_100_dart/lib/problem_003.dart
int largestPrimeFactor(int value) {
  int prime = 2, max = 1;
  while (prime <= value) {
    if (value % prime == 0) {
      max = prime;
      value = (value / prime).truncate();
    } else
      prime++;
  }
  return max;
}

File: project-euler-100-java/src/main/java/com/github/adrianomoreira/projecteuler100/Problem003.java
package com.github.adrianomoreira.projecteuler100;

public class Problem003 {

    public static long largestPrimeFactor(long value) {
        long prime = 2, max = 1;

        while (prime <= value) {
            if(value % prime == 0) {
                max = prime;
                value /= prime;
            } else prime++;
        }

        return max;
    }
}
